<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>970</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>9</y>
      <width>541</width>
      <height>451</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <item>
      <widget class="QCustomPlot" name="progressPlot" native="true"/>
     </item>
     <item>
      <widget class="QCustomPlot" name="pacPlot" native="true"/>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>351</width>
      <height>451</height>
     </rect>
    </property>
    <property name="title">
     <string>GroupBox</string>
    </property>
    <widget class="QComboBox" name="optEngine">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>330</y>
       <width>125</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <item>
      <property name="text">
       <string>Native + Eigen</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>VexCL</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ViennaCL</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>198</y>
       <width>70</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Iterations:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ntrainEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>132</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>10</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>66</y>
       <width>128</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Output dimension:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lambdaEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>231</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>1e-5</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>264</y>
       <width>73</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Batch size:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="gramEngine">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>297</y>
       <width>90</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <item>
      <property name="text">
       <string>Native + Eigen</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>VexCL</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ViennaCL</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>99</y>
       <width>92</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Test samples:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>297</y>
       <width>91</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Gram Engine:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="outdimEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>66</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>2</string>
     </property>
    </widget>
    <widget class="QComboBox" name="kernelCombo">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>165</y>
       <width>90</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <item>
      <property name="text">
       <string>Linear</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Gaussian</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Sine</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Cosine</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>33</y>
       <width>114</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Input dimension:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="indimEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>33</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>5</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>330</y>
       <width>77</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>OptEngine:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="nbatchEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>264</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
    <widget class="QLineEdit" name="ntestEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>99</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>10</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>132</y>
       <width>97</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Train samples:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="nitersEdit">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>198</y>
       <width>146</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>1000</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_6">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>360</y>
       <width>101</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Test X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_7">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>360</y>
       <width>100</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Test Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_8">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>360</y>
       <width>110</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Test YH</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>390</y>
       <width>106</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Train X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_3">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>390</y>
       <width>105</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Train Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_5">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>390</y>
       <width>98</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Alpha</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>420</y>
       <width>138</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Show Hyperplane</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>420</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Learn</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>49</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Kernel:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>58</width>
       <height>17</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Lambda:</string>
     </property>
    </widget>
    <zorder>optEngine</zorder>
    <zorder>label_6</zorder>
    <zorder>ntrainEdit</zorder>
    <zorder>label_2</zorder>
    <zorder>lambdaEdit</zorder>
    <zorder>label_8</zorder>
    <zorder>label_4</zorder>
    <zorder>gramEngine</zorder>
    <zorder>label_5</zorder>
    <zorder>label_9</zorder>
    <zorder>label_7</zorder>
    <zorder>outdimEdit</zorder>
    <zorder>kernelCombo</zorder>
    <zorder>label</zorder>
    <zorder>indimEdit</zorder>
    <zorder>label_10</zorder>
    <zorder>nbatchEdit</zorder>
    <zorder>ntestEdit</zorder>
    <zorder>label_3</zorder>
    <zorder>nitersEdit</zorder>
    <zorder>optEngine</zorder>
    <zorder>label_6</zorder>
    <zorder>ntrainEdit</zorder>
    <zorder>label_2</zorder>
    <zorder>lambdaEdit</zorder>
    <zorder>label_8</zorder>
    <zorder>gramEngine</zorder>
    <zorder>label_5</zorder>
    <zorder>label_9</zorder>
    <zorder>outdimEdit</zorder>
    <zorder>kernelCombo</zorder>
    <zorder>label</zorder>
    <zorder>indimEdit</zorder>
    <zorder>label_10</zorder>
    <zorder>nbatchEdit</zorder>
    <zorder>ntestEdit</zorder>
    <zorder>label_3</zorder>
    <zorder>nitersEdit</zorder>
    <zorder>pushButton_6</zorder>
    <zorder>pushButton_7</zorder>
    <zorder>pushButton_8</zorder>
    <zorder>pushButton_2</zorder>
    <zorder>pushButton_3</zorder>
    <zorder>pushButton_5</zorder>
    <zorder>pushButton_4</zorder>
    <zorder>pushButton</zorder>
    <zorder>label_7</zorder>
    <zorder>label_4</zorder>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>970</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionUnitTest">
   <property name="text">
    <string>UnitTest</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
